src/app/reporting/Engine.cpp

```diff
@@ -139,14 +150,27 @@ CHIP_ERROR Engine::BuildSingleReportDataAttributeDataList(ReportData::Builder &
             }
         }
     }
-    attributeDataList.EndOfAttributeDataList();
-    err = attributeDataList.GetError();
-
 exit:
+    if ((err == CHIP_ERROR_BUFFER_TOO_SMALL) || (err == CHIP_ERROR_NO_MEMORY))
+    {
+        ChipLogDetail(DataManagement, "<RE:Run> We cannot put more chunks into this report. Enable chunking.");
+        err = CHIP_NO_ERROR;
+    }
+
+    if (err == CHIP_NO_ERROR)
+    {
+        attributeDataList.EndOfAttributeDataList();
+        err = attributeDataList.GetError();
+    }
+
+    ChipLogDetail(DataManagement, "----err %" PRId32 " remaining size %" PRIu32, err.AsInteger(),
+                  aReportDataBuilder.GetWriter()->GetRemainingFreeLength());
+
     if (attributeClean || err != CHIP_NO_ERROR)
```

Comment:
I can't figure out what this `attributeClean` variable does. Can you please provide some comments/explanations for it?

Reply:
The attributeClean variable is used to indicate if we have generated any meanful reports.